packages/web3-shared/base/src/specs/index.ts

```diff
@@ -604,17 +609,31 @@ export interface NonFungibleTokenSecurity {}
 
  export interface Result<ChainId> {
     pluginID: NetworkPluginID
-    chainId?: ChainId
     type: SearchResultType
     keyword: string
+    chainId?: ChainId
+    id?: string
+    name?: string
+    rank?: number
+    address?: string
+    source?: SourceType
+    logoURL?: string
+    symbol?: string
+}
+export interface EOAResult<ChainId> extends Result<ChainId> {
+    type: SearchResultType.EOA
+    domain?: string
+    address: string
 }
 
-export interface AddressResult<ChainId> extends Result<ChainId> {
-    type: SearchResultType.Address
+export interface CAResult<ChainId> extends Result<ChainId> {
```

Comment:
This `CAResult` duplicate with `FungibleTokenResult` or `NonFungibleTokenResult`.

Reply:
The `DSearchAPI` should take responsibility for returning what a contract address belongs to. A rough `CAResult` is meanless for UI.